tests/suites/test_suite_gcm.aes128_de.data

```diff
@@ -670,6 +670,10 @@ AES-GCM NIST Validation (AES-128,128,1024,1024,32) #2 [#2]
 depends_on:MBEDTLS_AES_C
 gcm_decrypt_and_verify:MBEDTLS_CIPHER_ID_AES:"659b9e729d12f68b73fdc2f7260ab114":"fd0732a38224c3f16f58de3a7f333da2ecdb6eec92b469544a891966dd4f8fb64a711a793f1ef6a90e49765eacaccdd8cc438c2b57c51902d27a82ee4f24925a864a9513a74e734ddbf77204a99a3c0060fcfbaccae48fe509bc95c3d6e1b1592889c489801265715e6e4355a45357ce467c1caa2f1c3071bd3a9168a7d223e3":"459df18e2dfbd66d6ad04978432a6d97":"ee0b0b52a729c45b899cc924f46eb1908e55aaaeeaa0c4cdaacf57948a7993a6debd7b6cd7aa426dc3b3b6f56522ba3d5700a820b1697b8170bad9ca7caf1050f13d54fb1ddeb111086cb650e1c5f4a14b6a927205a83bf49f357576fd0f884a83b068154352076a6e36a5369436d2c8351f3e6bfec65b4816e3eb3f144ed7f9":32:"8e5a6a79":"FAIL":"":0
 
+AES-GCM NIST Validation (AES-128,128,0,256,128) #0 [#0]
```

Comment:
There are already plenty of test cases with empty ciphertext and non-empty AD, so I don't think the two new test cases are useful.

----

In case they do turn out to be useful:

Please don't add `[#N]` annotations on test cases and avoid adding `#N`. `#N` is a bad historical habit of using numbers rather than useful descriptions. `[#N]` annotations were added automatically when test cases had identical descriptions which made the test reporting hard to use. In both cases, we eventually want to replace those by meaningful descriptions (except in rare cases where we're just testing crypto with some randomish data and there's nothing informative about the difference between randomish data 1 and randomish data 2). If the goal of this test case is to test with empty ciphertext and non-empty AD, the test description should say so. 

Did this test data actually come from a NIST validation suite? I can't find `ff7ad50df57d635839acf143af43f002` in 
http://csrc.nist.gov/groups/STM/cavp/documents/mac/gcmtestvectors.zip .

Reply:
I was not aware we should pick test vectors from some special place. I guess I'll remove them anyways, since (as you pointed it out) they don't cover increase the coverage.

Reply:
We don't have to pick test vectors from a special place, but 1. we do need to know why we believe that they are correct and 2. if the description says “NIST Validation” then they should indeed come from a NIST validation suite.